src/zjs_fs.c

```diff
@@ -413,17 +421,21 @@ static jerry_value_t zjs_read(const jerry_value_t function_obj,
         return zjs_error("offset + length overflows buffer");
     }
 
-    if (handle->mode == MODE_A_PLUS) {
+    // a+ or read from current position (position == null)
+    if ((handle->mode == MODE_A_PLUS) || from_cur) {
```

Comment:
I don't think you want the from_cur part here. If you're not in A_PLUS, rpos isn't needed, just trust the actually current position the OS is tracking.